File: O/source_code_10.cpp
#include <iostream>
#include <cmath>
using namespace std;

float f(float x)
{
    return x * x - 4 * x - 10;
}

int main()
{
    float a = -2, b = -1.5, c, tolerance = 0.0001;

    if (f(a) * f(b) >= 0)
    {
        cout << "Invalid interval. f(a) and f(b) must have opposite signs." << endl;
        return 1;
    }

    while ((b - a) >= tolerance)
    {
        c = (a + b) / 2;
        if (f(c) == 0.0)
            break;
        else if (f(c) * f(a) < 0)
            b = c;
        else
            a = c;
    }

    cout << "The root is approximately: " << c << endl;
    return 0;
}

File: O/source_code_10.py
def f(x): return x**2 - 4*x - 10

a, b = -2, -1.5
tolerance = 0.0001

while (b - a) >= tolerance:
    c = (a + b) / 2
    if f(c) == 0:
        break
    elif f(c) * f(a) < 0:
        b = c
    else:
        a = c

print("Root using Bisection:", c)
